packages/react-devtools-extensions/src/main.js

```diff
@@ -242,21 +250,59 @@ function createPanelIfReactLoaded() {
           // This helper function allows the extension to request files to be fetched
           // by the content script (running in the page) to increase the likelihood of a cache hit.
           fetchFileWithCaching = url => {
-            const event = cachedNetworkEvents.get(url);
-            if (event != null) {
-              // If this resource has already been cached locally,
-              // skip the network queue (which might not be a cache hit anyway)
-              // and just use the cached response.
-              return new Promise(resolve => {
-                event.getContent(content => resolve(content));
-              });
+            if (__DEBUG__) {
+              console.log('[main] fetchFileWithCaching() for ', url);
             }
+            return new Promise((resolve, reject) => {
+              const request = cachedRequests.get(url);
+              let cachedContent = null;
+              if (request != null) {
+                if (__DEBUG__) {
+                  console.log(
+                    '[main] fetchFileWithCaching() Found a cached network request for ',
+                    url,
+                  );
+                }
+
+                request.getContent(content => {
```

Comment:
Is this really called synchronously? 😮

Reply:
It would have to be or the check below wouldn't make sense...but did you confirm that?

Reply:
oops, no, I was just being dumb and didn't properly catch this in my test (since there aren't any runtime errors anymore).

good catch! will fix now